animal_crossing.py

```diff
@@ -0,0 +1,70 @@
+import re
+import sqlite3
+
+import config
+
+if config.bot.acnh_db is not None:
+	db = sqlite3.connect(config.bot.acnh_db)
+	db.row_factory = sqlite3.Row
+
+def friend_code(cmd):
+	if not cmd.args:
+		_user_list_all(cmd)
+		return
+
+	split = cmd.args.split(' ', 1)
+	subcmd = split[0]
+	if subcmd == 'set':
+		_user_upsert_friend_code(cmd, split[1])
+	elif subcmd == 'remove':
+		_user_remove(cmd)
+	else:
+		_user_find(cmd, split[0])
+
+def _user_upsert_friend_code(cmd, code):
+	pat = re.compile(r'SW-\d{4}-\d{4}-\d{4}')
```

Comment:
did you mean to compile the regex outside the function?
you can directly just `re.match(r'...', code)` and it will cache the compiled re. but this way, you're losing the benefit of the compilation

Reply:
yes, I did :sweat: